### ISSUE TITLE
Enhancements Needed

### ISSUE THREAD

**Issue by dlowrey** [NilsG-S/team-null-back-end#33]
- ~Send Receipt when patient pays invoice or penalty~ done
- ~cron job to send penalty email every day at 9~
- ~modifyPaymentById needed for when patient logs on with Invoice Id to pay invoice (also need to switch receipt sending to here when it is added)~
- ~cron job to update all appointments in database that arent completed to completed = 2 at 9~
- ~transfer all MySQL Events to Node.js cron jobs? (daily/monthly reports)~

**Comment by NilsG-S** [comment]
@dlowrey 
I was assuming that a penalty payment would be basically the same as a regular payment, I was more thinking in terms of maybe emailing people when they have a penalty payment. What do you think?

**Comment by dlowrey** [comment]
@NilsG-S 
Yeah I was thinking along those lines as well. I am not sure how I am going to know when to send the email though.

**Comment by NilsG-S** [comment]
Use [node-cron](https://www.npmjs.com/package/node-cron). [Here's](http://stackoverflow.com/questions/33366847/how-to-setup-node-schedule-for-every-day-at-12am) an example of what you want.

**Comment by dlowrey** [comment]
@NilsG-S 
Beautiful, that solves all of my problems. I will do those tomorrow.

**Comment by dlowrey** [comment]
@NilsG-S what do you think about the latest bullet point

**Comment by NilsG-S** [comment]
@dlowrey Like I said, I think it might be cleaner, but I also don't have any direct experience with this kind of thing. I assume by your question that you're considering it?

**Comment by dlowrey** [comment]
@NilsG-S all events (daily report generation, monthly report generation) are now handled with cron jobs.
